src/main/java/io/github/nucleuspowered/nucleus/modules/item/commands/RepairCommand.java

```diff
@@ -4,46 +4,274 @@
-public class RepairCommand extends AbstractCommand.SimpleTargetOtherPlayer {
-
-    @Override protected CommandResult executeWithPlayer(CommandSource src, Player pl, CommandContext args, boolean isSelf) throws Exception {
-        if (pl.getItemInHand(HandTypes.MAIN_HAND).isPresent()) {
-            ItemStack stack = pl.getItemInHand(HandTypes.MAIN_HAND).get();
-            if (stack.get(DurabilityData.class).isPresent()) {
-                DurabilityData durabilityData = stack.get(DurabilityData.class).get();
-                DataTransactionResult transactionResult = stack.offer(Keys.ITEM_DURABILITY, durabilityData.durability().getMaxValue());
-                if (transactionResult.isSuccessful()) {
-                    pl.setItemInHand(HandTypes.MAIN_HAND, stack);
-                    src.sendMessage(plugin.getMessageProvider().getTextMessageWithFormat("command.repair.success", pl.getName()));
-                    return CommandResult.success();
+public class RepairCommand extends AbstractCommand<Player> implements Reloadable {
+
+    private boolean whitelist = false;
+    private List<ItemType> restrictions = new ArrayList<>();
+
+    @Override public void onReload() throws Exception {
+        this.whitelist = Nucleus.getNucleus().getInternalServiceManager().getServiceUnchecked(ItemConfigAdapter.class)
+                .getNodeOrDefault().getRepairConfig().isWhitelist();
+        this.restrictions = Nucleus.getNucleus().getInternalServiceManager().getServiceUnchecked(ItemConfigAdapter.class)
+                .getNodeOrDefault().getRepairConfig().getRestrictions();
+    }
+
+    @Override public CommandElement[] getArguments() {
+        return new CommandElement[]{
+                GenericArguments.flags()
+                        .flag("m", "-mainhand")
+                        .permissionFlag(permissions.getPermissionWithSuffix("all"), "a", "-all")
+                        .permissionFlag(permissions.getPermissionWithSuffix("hotbar"), "h", "-hotbar")
+                        .permissionFlag(permissions.getPermissionWithSuffix("equip"), "e", "-equip")
+                        .permissionFlag(permissions.getPermissionWithSuffix("offhand"), "o", "-offhand")
+                        .buildWith(GenericArguments.none())
+        };
+    }
+
+    @Override protected CommandResult executeCommand(Player pl, CommandContext args) throws Exception {
+        EnumMap<ResultType, Integer> resultCount = new EnumMap(ResultType.class) {{
+            put(ResultType.SUCCESS, 0);
+            put(ResultType.ERROR, 0);
+            put(ResultType.NO_DURABILITY, 0);
+            put(ResultType.RESTRICTED, 0);
+        }};
+        EnumMap<ResultType, ItemStackSnapshot> lastItem = new EnumMap<>(ResultType.class);
+
+        String location = "inventory";
+        if (args.hasAny("a")) {
+            for (Inventory slot : pl.getInventory().slots()) {
+                if (slot.peek().isPresent() && !slot.peek().get().isEmpty()) {
```

Comment:
Could we not take lines 78 to 86 and wrap it up in a function that we pass the maps into?

```java
private void repairSlot(Inventory slot, EnumMap<ResultType, Integer> resultCount, EnumMap<ResultType, ItemStackSnapshot> lastItem) {
    // code here
}
```

The argument references are passed, so it'll update the original map.

Reply:
I always forget there's some trickery involved that makes java no always pass by value. I didn't test this change yet but I went ahead and made it.

Reply:
It's a weird one, Java is pass reference by value - so you can provide an object and change the internals of that object, but it won't return a new instance.